### PR TITLE
ci: build conda env in GHA

### PR DESCRIPTION
**Pull request by andrzejnovak**
This adds a GHA workflow that 
- checks out the conda env [for 3.6 and 3.7 python]
- compiles the submodules
- runs the example subpackage training

CI passed in forked branch https://github.com/andrzejnovak/DeepJetCore/runs/1520435913?check_suite_focus=true should work on master after the GHA workflow is initiated i.e. has the `.yml` file

### CODE REVIEW

File: conda_env.yml [hunk 1]
```diff
@@ -278,4 +278,4 @@ dependencies:
     - uproot3==3.14.1
     - uproot3-methods==0.10.0
     - widgetsnbextension==3.5.1
-prefix: /beegfs/desy/user/anovak/miniconda3/envs/DLtest
+prefix: /beegfs/desy/user/anovak/miniconda3/envs/DJC2
```
Comment (jkiesele) [inline conda_env.yml]: I think this line should be removed entirely

**Comment by jkiesele** [comment]
Excellent! Thanks a lot.
This is a great starting point. I think we can now add more and more CI checks as we go!

File: conda_env.yml (hunk 1 above)
Comment (andrzejnovak) [reply conda_env.yml]: Good catch. I suppose that gets created to show provenance, but no needed here.

**Comment by andrzejnovak** [comment]
If I have time, I can try to remake this as well with a less constrained conda spec -> the CI should then alert you when some package breaks the functionality.

You should also be able to load docker in GHA, so you could similarly test your other environment. But I don't think I have the time to look into that myself.

**Comment by andrzejnovak** [comment]
Ok, the action won't run in this PR, because it's coming from a fork and for safety reasons that doesn't happen unless the `.yml` is already in the target repo. 

Should be safe to merge after the run in the PR branch finishes https://github.com/andrzejnovak/DeepJetCore/actions/runs/408710398

**Comment by andrzejnovak** [comment]
@jkiesele should be good to merge

**Comment by andrzejnovak** [comment]
@jkiesele I've merged what was gonna be next PR into this one.
- changed the conda env to only have minimally needed recs
- modded the makefile to automatically get the right boost files for the python installed into conda
- action now runs a limited matrix - checks for python 3.6 (your docker default) and 3.7
- I didn't include 3.8 because there were built errors 

- original conda_env is kept as a backup file, in case the non-specified versions break in the future, so a user has something to fallback while it's fixed

**Comment by andrzejnovak** [comment]
On a side note, I've had to limit the h5py<3 version for TF compatibility reasons. IIUC this is not specified in the docker file, so I am kinda surprised there weren't any problems.

**Comment by jkiesele** [comment]
Thanks a lot! Excellent.
It can be that due to the strict sequential installing of h5py and then TF, it downgraded it automatically.

**Pull request merged**